### PR TITLE
ci: add shfmt workflow

### LINKED ISSUE #3588: Lint shell scripts with shfmt
Lint all ockam shell script with [shfmt](https://github.com/mvdan/sh) in CI. Requires #3587

### PR DESCRIPTION
**Pull request by elpiel**
Resolves #3588 

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

File: .github/workflows/shfmt.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: Shfmt - Lint
+    runs-on: ubuntu-20.04
+    container:
+      image: mvdan/shfmt
```
Comment (mrinalwadhwa) [inline .github/workflows/shfmt.yml]: We like to pin all dependencies. Let's pin this container to its current sha

**Review by mrinalwadhwa — CHANGES_REQUESTED** [review]
@elpiel This looks great. Thank you for spending time on this.
Left a small suggestion, after which we can merge.

File: .github/workflows/shfmt.yml (hunk 1 above)
Comment (elpiel) [reply .github/workflows/shfmt.yml]: Ahh, yes. I was thinking of that and I noticed you do it. Will do that first thing in the morning

Comment (mrinalwadhwa) [reply .github/workflows/shfmt.yml]: awesome, thank you

Comment (elpiel) [reply .github/workflows/shfmt.yml]: Done.

**Comment by elpiel** [comment]
> This workflow is awaiting approval from a maintainer in https://github.com/build-trust/ockam/pull/3698

File: .github/workflows/shfmt.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  lint:
+    name: Shfmt - Lint
+    runs-on: ubuntu-20.04
+    container:
+      image: mvdan/shfmt@sha256:93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+    steps:
+      - name: Check out ockam repository
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      - name: Run Shfmt
+        run: shfmt -d
```
Comment (metaclips) [inline .github/workflows/shfmt.yml]: This call blocks, shouldn't we indicate files to be formatted?

Comment (elpiel) [reply .github/workflows/shfmt.yml]: I will take a look, but maybe you can approve the run of the CI so we can see if it runs?

Comment (elpiel) [reply .github/workflows/shfmt.yml]: I've added the same fix as in `shellcheck` using `find`.

**Review by metaclips — DISMISSED** [review]
Thanks @elpiel

**Comment by elpiel** [comment]
ohhhh no, it still gets the `tail` not found error when the docker container is run...

**Comment by metaclips** [comment]
@elpiel I think it gets easier if we add shmft to the builder image

**Comment by elpiel** [comment]
> @elpiel I think it gets easier if we add shmft to the builder image

Since it's already installed in the builder image I've just changed to use the builder instead of `shfmt` docker image.

Ready to run and review again :)

File: .github/workflows/shfmt.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  lint:
+    name: Shfmt - Lint
+    runs-on: ubuntu-20.04
+    container:
+      image: ghcr.io/build-trust/ockam-builder@sha256:e43dd94652096b03cc472a3c709c7335e8b166cab77b7a7b56f88fa38f3d24cc
```
Comment (metaclips) [inline .github/workflows/shfmt.yml]: Let's use the latest image SHA `sha256:9598b7bc30bb23e9e64cd95cb96a30af9695e6a16ce1834e6c17c3c62bc81c2e` the current doesn't have `shfmt` installed.

File: .github/workflows/shfmt.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+      - name: Check out ockam repository
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8
+      - name: Run Shfmt on all ''.sh' files
```
Comment (metaclips) [inline .github/workflows/shfmt.yml]: Not sure if the quote format was intended

**Review by metaclips — DISMISSED** [review]
Thanks @elpiel. Little changes for a merge

File: .github/workflows/shfmt.yml (hunk 3 above)
Comment (elpiel) [reply .github/workflows/shfmt.yml]: Fixed.

File: .github/workflows/shfmt.yml (hunk 4 above)
Comment (elpiel) [reply .github/workflows/shfmt.yml]: That was unintended and it's fixed now.

**Review by mrinalwadhwa — APPROVED** [review]

**Comment by mrinalwadhwa** [comment]
Thank you @elpiel for another awesome contribution 🥳🎊

**Pull request merged**